sematic/config/settings.py

```diff
@@ -2,15 +2,194 @@
+    server, etc).
+
+    Instances of this classes also contain the cached settings value.
+
+    Parameters
+    ----------
+    file_name: str
+        Name of the YAML file containing settings value for this scope.
+
+    cli_command: str
+        Command to set/get settings values for this scope.
+
+    vars: Type[AbstractSettingsVar]
+        Enum of available settings for this scope
+    """
+
+    file_name: str
+    cli_command: str
+    vars: Type[AbstractSettingsVar]
+    _settings: Optional[Settings] = field(init=False, default=None)
+    _active_settings: Optional[ProfileSettings] = field(init=False, default=None)
+
+    @property
+    def settings_file_path(self) -> str:
+        """
+        Full path of settings YAML file.
+        """
+        return os.path.join(get_config_dir(), self.file_name)
+
+    def get_settings(self) -> Settings:
+        """
+        Get and cache content of settings file filtered to available settings
+        for this scope.
+        """
+        if self._settings is None:
+            self._settings = _load_settings(self.settings_file_path, self.vars)
+
+        return self._settings
+
+    def get_active_settings(self) -> ProfileSettings:
+        """
+        Get and cache effective settings value for current profile (default).
+
+        This also applies env var overrides.
+        """
+        if self._active_settings is None:
+            self._active_settings = self.get_settings()[DEFAULT_PROFILE]
+            _apply_env_var_overrides(self._active_settings, self.vars)
+
+        return self._active_settings
+
+    def get_active_settings_as_dict(self) -> Dict[str, str]:
+        """
+        Active settings as a dictionary.
+        """
+        return {var.value: value for var, value in self.get_active_settings().items()}
+
+    def get_setting(self, var: AbstractSettingsVar, *args) -> str:
+        """
+        Retrieves and returns the specified settings value, with environment
+        override.
+
+        Loads and returns the specified settings value. If it does not exist, it
+        falls back on the first optional vararg as a default value. If that does
+        not exist, it raises.
+        """
+        value = self.get_active_settings().get(var)
+
+        if value is not None:
+            return str(value)
+
+        if len(args) >= 1:
+            return args[0]
+
+        raise MissingSettingsError(var, self.cli_command)
+
+    def set_setting(self, var: AbstractSettingsVar, value: str) -> None:
+        """
+        Set value for setting.
+
+        Parameters
+        ----------
+        var: AbstractSettingsVar
+            setting to set
+        value: str
+            value to set
+        """
+        if var not in self.vars:
+            raise ValueError(f"Unknown setting {var.value}")
+
+        settings = self.get_settings()
+
+        settings[DEFAULT_PROFILE][var] = value
+
+        self.save_settings()
```

Comment:
This code works by using `self._settings` through a side channel, as `self.save_settings()` calls `self.get_settings()` again.

Please make the code flow explicit. Same comment applies for `delete_setting()`.
```suggestion
        settings = self.get_settings()

        settings[DEFAULT_PROFILE][var] = value

        self.save_settings(settings)
```

Reply:
Done.